cl_sii/dte/parse.py

```diff
@@ -538,6 +556,93 @@ def _text_strip_or_raise(xml_em: XmlElement) -> str:
+    """
+    Reusable Pydantic validator for fields of type :class:`Rut`.
+    """
+    if isinstance(v, str):
+        v = Rut(value=v, validate_dv=False)  # Raises ValueError if invalid.
+    return v
+
+
+class _DteXmlReferenciaParser(pydantic.BaseModel):
+    """
+    Parser for ``/Documento/Referencia``.
+    """
+
+    class Config:
+        allow_mutation = False
+        anystr_strip_whitespace = True
+        arbitrary_types_allowed = True
+        extra = pydantic.Extra.forbid
+
+    ###########################################################################
+    # Fields
+    ###########################################################################
+
+    numero_linea_ref: int
+    tipo_documento_ref: str
+    ind_global: Optional[int]
+    folio_ref: str
+    rut_otro: Optional[Rut]
+    fecha_ref: date
+    codigo_ref: Optional[constants.CodigoReferencia]
+    razon_ref: Optional[str]
+
+    ###########################################################################
+    # Custom Methods
+    ###########################################################################
+
+    @classmethod
+    def parse_xml(cls, xml_doc: XmlElement) -> _DteXmlReferenciaParser:
+        aec_dict = cls.parse_xml_to_dict(xml_doc)
+        return cls.parse_obj(aec_dict)
+
+    def as_dte_xml_referencia(self) -> data_models.DteXmlReferencia:
+        return data_models.DteXmlReferencia(
+            numero_linea_ref=self.numero_linea_ref,
+            tipo_documento_ref=self.tipo_documento_ref,
+            ind_global=self.ind_global,
+            folio_ref=self.folio_ref,
+            rut_otro=self.rut_otro,
+            fecha_ref=self.fecha_ref,
+            codigo_ref=self.codigo_ref,
+            razon_ref=self.razon_ref,
+        )
+
+    @staticmethod
+    def parse_xml_to_dict(xml_em: XmlElement) -> Mapping[str, object]:
+        """
+        Parse XML element and return a dictionary.
+        """
+        # XPath: /Documento/Referencia
+        return dict(
+            numero_linea_ref=xml_em.findtext('sii-dte:NroLinRef', namespaces=DTE_XMLNS_MAP),
+            tipo_documento_ref=xml_em.findtext('sii-dte:TpoDocRef', namespaces=DTE_XMLNS_MAP),
+            ind_global=xml_em.findtext('sii-dte:IndGlobal', namespaces=DTE_XMLNS_MAP) or None,
+            folio_ref=xml_em.findtext('sii-dte:FolioRef', namespaces=DTE_XMLNS_MAP),
+            rut_otro=xml_em.findtext('sii-dte:RUTOtr', namespaces=DTE_XMLNS_MAP) or None,
+            fecha_ref=xml_em.findtext('sii-dte:FchRef', namespaces=DTE_XMLNS_MAP),
+            codigo_ref=xml_em.findtext('sii-dte:CodRef', namespaces=DTE_XMLNS_MAP) or None,
+            razon_ref=xml_em.findtext('sii-dte:RazonRef', namespaces=DTE_XMLNS_MAP) or None,
+        )
+
+    ###########################################################################
+    # Validators
+    ###########################################################################
+
+    _validate_rut = pydantic.validator(  # type: ignore[pydantic-field]
```

Comment:
This `# type: ignore` is no longer needed

```suggestion
    _validate_rut = pydantic.validator(
```